salt/metalk8s/orchestrate/deploy_new_node.sls

```diff
@@ -0,0 +1,65 @@
+{%- if pillar['bootstrap_id'] %}
+{%-   set control_plane_ips = salt.saltutil.runner('mine.get', tgt=pillar['bootstrap_id'], fun='control_plane_ip') %}
+{%- else %}
+{%-   set control_plane_ips = {} %}
+{%- endif %}
+
+{%- if pillar['bootstrap_id'] in control_plane_ips.keys() and control_plane_ips[pillar['bootstrap_id']] %}
+{%-   set control_plane_ip = control_plane_ips[pillar['bootstrap_id']] %}
+{%- else %}
+{%-   set control_plane_ip = 'localhost' %}
+{%- endif %}
+
+
+Deploy salt-minion on a new node:
+  salt.state:
+    - ssh: true
+    - roster: kubernetes
+    - tgt: {{ pillar['node_name'] }}
+    - saltenv: {{ saltenv }}
+    - sls:
+      - metalk8s.roles.salt-minion
+    - pillar:
+        repo:
+          host: {{ control_plane_ip }}
+        salt:
+          master:
+            host: {{ control_plane_ip }}
+
+Set grains:
+  salt.state:
+    - tgt: {{ pillar['node_name'] }}
+    - sls:
+      - metalk8s.node.grains
+    - require:
+      - salt: Deploy salt-minion on a new node
+
+Refresh the mine:
+  salt.function:
+    - name: mine.update
+    - tgt: '*'
+
+Run the highstate:
+  salt.state:
+    - tgt: {{ pillar['node_name'] }}
+    - highstate: True
+    - require:
+      - salt: Set grains
+      - salt: Refresh the mine
+
+{%- if 'etcd' in pillar.get('nodes', {}).get(pillar['node_name'], {}).get('roles', []) %}
+
+  {%- set node_ip = grains['metalk8s']['control_plane_ip'] %}
```

Comment:
This will be the master's `control_plane_ip`. How would that work then?

Reply:
Yes I agree I saw that also but that was like that before this PR in `deploy_etcd..` orchestrate so I just copy/paste this

Reply:
#1017